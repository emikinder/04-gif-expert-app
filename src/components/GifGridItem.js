import React from 'react';
import PropTypes from 'prop-types';

const GifGridItem = ({ title, image }) => {
	return (
		<div className='card animate__animated animate__zoomIn'>
			<img src={image} alt={title} />
			<p>{title}</p>
		</div> 
	);
};

GifGridItem.propTypes = {
	title: PropTypes.string.isRequired,
	image: PropTypes.string.isRequired
}

export default GifGridItem;
