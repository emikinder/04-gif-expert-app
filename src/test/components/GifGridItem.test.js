import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe('GifGridItem tests', () => {
	const title = 'Titulo';
	const image = 'https://localhost/img.png';
	const wrapper = shallow(<GifGridItem title={title} image={image} />);

	test('debería mostrar el componente correctamente', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('debería tener un párrafo con el title', () => {
		const p = wrapper.find('p');
		expect(p.text().trim()).toBe(title);
	});

	test('debería tener una imagen con src y alt', () => {
		const img = wrapper.find('img');
		const { src, alt } = img.props();
		// expect( img.prop('src') ).toBe(image);
		expect(src).toBe(image);
		expect(alt).toBe(title);
	});

	test('debería contener la clase zoomIn', () => {
		const div = wrapper.find('div');
		const className = div.prop('className');
		// expect(className).toContain('animate__zoomIn')
		expect(className.includes('animate__zoomIn')).toBe(true);
	});
});
