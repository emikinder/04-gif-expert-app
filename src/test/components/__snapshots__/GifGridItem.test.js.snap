// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem tests deber√≠a mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__zoomIn"
>
  <img
    alt="Titulo"
    src="https://localhost/img.png"
  />
  <p>
    Titulo
  </p>
</div>
`;
